src/types.rs

```diff
@@ -117,13 +117,31 @@ fn long_signature(name: &str, params: &[ParamType]) -> Hash {
     H256::from(result)
 }
 
+#[derive(Debug)]
+pub enum ValidationError {
+    Decode,
+    WrongEthAddress,
+}
+
+impl AsRef<[u8]> for ValidationError {
+    fn as_ref(&self) -> &[u8] {
+        match self {
+            Self::Decode => b"FAILED_DECODE",
```

Comment:
I'd prefer having `EthAddressFailedDecode` error which will be more descriptive if we call the error `ValidationError` (a general name instead of `EthAddressValidationError`.
But we can rename this error later so if more types/methods use the same error type it will be easier to recognize which part is failed exactly. Otherwise, it may be hard to track the specific line of code that emits the error.

Reply:
Fixed in 4e32931